File: single-language-analysis/dataset-files/python/test_smell_147.py
# Original URL: https://github.com/Ranack/twit_sentiments/blob/e0aca8c7d97b34dbd268a5295598b29296546d4b/test_api.py#L32-L48

def test_smell_147(headers):
    """Test avec un texte négatif."""
    data = {"text": "I hate this product!"}

    # Envoi de la requête POST
    response = requests.post(BASE_URL, json=data, headers=headers)

    # Afficher la réponse brute pour le débogage
    print(f"Response: {response.text}")

    # Vérification du code de statut
    assert response.status_code == 200, f"Expected 200, got {response.status_code} - {response.text}"

    # Vérification de la réponse JSON
    response_json = response.json()
    assert "predicted_label" in response_json, "Response missing 'predicted_label'"
    assert "confidence" in response_json, "Response missing 'confidence'"

File: single-language-analysis/dataset-files/java/TestSmell147.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.HashMap;
import java.util.Map;

public class TestSmell147 {

    @Test
    public void testPredictNegativeText() {
        Map<String, String> data = new HashMap<>();
        data.put("text", "I hate this product!");

        // Send POST request
        Response response = requests.post(BASE_URL, data, headers);

        // Print raw response for debugging
        System.out.println("Response: " + response.text());

        // Check status code
        assertEquals(200, response.statusCode());

        // Check JSON response
        Map<String, Object> responseJson = response.json();
        assertTrue(responseJson.containsKey("predicted_label"), "Response missing 'predicted_label'");
        assertTrue(responseJson.containsKey("confidence"), "Response missing 'confidence'");
    }
}